lib/openssh/sshd.go

```diff
@@ -0,0 +1,243 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openssh
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/trace"
+)
+
+const sshdBinary = "sshd"
+
+var (
+	// SSHDConfigPath is the path to write teleport specific SSHD config options
+	SSHDConfigPath = filepath.Join(defaults.DataDir, "sshd.conf")
+	// SSHDKeysDir is the path to the Teleport openssh keys
+	SSHDKeysDir = filepath.Join(defaults.DataDir, "openssh")
+
+	// sshdConfigInclude is Include directive added to the sshd_config file
+	sshdConfigInclude = "Include " + SSHDConfigPath
+)
+
+const DefualtRestartCommand = "systemctl restart sshd"
+
+const (
+	// TeleportKey is the name the OpenSSH private key
+	TeleportKey = "ssh_host_teleport_key"
+	// TeleportKey is the name the OpenSSH cert
+	TeleportCert = "ssh-host-teleport-cert.pub"
```

Comment:
I'm not sure if there's some convention regarding where to store host certificates, but if we extrapolate from `.ssh/id_rsa`:

```suggestion
	TeleportCert = "ssh_host_teleport_key-cert.pub"
```

or maybe just

```suggestion
	TeleportCert = TeleportKey + "-cert.pub"
```